### ISSUE TITLE
An error was reported in next.js project   

### ISSUE THREAD

**Issue by cl510804194** [solana-labs/wallet-adapter#153]
i am not sure this is a bug
but when i use like this
`import { useWallet, ConnectionProvider, WalletProvider } from '@solana/wallet-adapter-react'`,
then throw an error ` Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/oo/Documents/work/flux-interface/node_modules/@solana/wallet-adapter-base/lib/adapter' imported from /Users/oo/Documents/work/flux-interface/node_modules/@solana/wallet-adapter-base/lib/index.js`

my next.js version is `^11.1.2`
and   "@solana/wallet-adapter-react": "^0.13.1"

**Comment by QZIG** [comment]
Hi !

check nextjs-starter, this file :
[https://github.com/solana-labs/wallet-adapter/blob/master/packages/starter/nextjs-starter/next.config.js](url)
Just be sure about installed and uninstalled packages

"@solana/wallet-adapter-base": "^0.7.0",
"@solana/wallet-adapter-react": "^0.13.1",
"@solana/wallet-adapter-wallets": "^0.11.2",
"@solana/web3.js": "^1.30.2",

This is my next.config.js :
`/* eslint-disable @typescript-eslint/no-var-requires */
const withTM = require("next-transpile-modules")([
  "@blocto/sdk",
  "@project-serum/sol-wallet-adapter",
  "@solana/wallet-adapter-base",
  "@solana/wallet-adapter-react",
  "@solana/wallet-adapter-bitkeep",
  "@solana/wallet-adapter-bitpie",
  "@solana/wallet-adapter-blocto",
  "@solana/wallet-adapter-clover",
  "@solana/wallet-adapter-coin98",
  "@solana/wallet-adapter-coinhub",
  "@solana/wallet-adapter-ledger",
  "@solana/wallet-adapter-mathwallet",
  "@solana/wallet-adapter-phantom",
  "@solana/wallet-adapter-safepal",
  "@solana/wallet-adapter-slope",
  "@solana/wallet-adapter-solflare",
  "@solana/wallet-adapter-sollet",
  "@solana/wallet-adapter-solong",
  "@solana/wallet-adapter-tokenpocket",
  "@solana/wallet-adapter-torus",
  "@solana/wallet-adapter-wallets",
]);

/** @type {import('next').NextConfig} */
module.exports = withTM({
  reactStrictMode: true,
  webpack5: true,
});`

**Comment by cl510804194** [comment]
thanks a lot

**Comment by QZIG** [comment]
just copy past or go through code ;)

**Comment by cl510804194** [comment]
thanks very much

**Comment by jjmnde** [comment]
Hi! The next.config.js solved the problem mentioned for me, but now I am getting
```
TypeError: (0 , _solana_wallet_adapter_wallets__WEBPACK_IMPORTED_MODULE_4__.getSolletWallet) is not a function
```

for  ` const wallets = useMemo(() => [getSolletWallet({ network })], [network]);`

I am using the same packages
```
    "@solana/wallet-adapter-base": "^0.7.0",
    "@solana/wallet-adapter-react": "^0.13.1",
    "@solana/wallet-adapter-wallets": "^0.11.2",
    "@solana/web3.js": "^1.30.2",
```

**Comment by jordansexton** [comment]
@jjmnde That suggests you aren't importing `getSolletWallet` from the wallets package. It generally helps to see the full code that's producing the error instead of just the error.

**Comment by jjmnde** [comment]
> @jjmnde That suggests you aren't importing `getSolletWallet` from the wallets package. It generally helps to see the full code that's producing the error instead of just the error.

I am using the code from the example
```
import { WalletAdapterNetwork } from "@solana/wallet-adapter-base";
import {
  ConnectionProvider,
  WalletProvider,
} from "@solana/wallet-adapter-react";
import { getSolletWallet } from "@solana/wallet-adapter-wallets";
import { clusterApiUrl } from "@solana/web3.js";
import { FC, ReactNode, useMemo } from "react";

export const WalletWrapper: FC<{ children: ReactNode }> = ({ children }) => {
  // Can be set to 'devnet', 'testnet', or 'mainnet-beta'
  const network = WalletAdapterNetwork.Devnet;

  // You can also provide a custom RPC endpoint
  const endpoint = useMemo(() => clusterApiUrl(network), [network]);

  // @solana/wallet-adapter-wallets includes all the adapters but supports tree shaking --
  // Only the wallets you configure here will be compiled into your application
  const wallets = useMemo(() => [getSolletWallet({ network })], [network]);

  return (
    <ConnectionProvider endpoint={endpoint}>
      <WalletProvider wallets={wallets} autoConnect>
        {children}
      </WalletProvider>
    </ConnectionProvider>
  );
};
```

**Comment by jordansexton** [comment]
Thanks @jjmnde, can you post a repo with a reproduction of the issue?

Sollet seems to work on the example here: https://solana-labs.github.io/wallet-adapter/example/

I don't know what the issue could be yet, but I can take a look at your project.

**Comment by dylf** [comment]
```js
import { WalletAdapterNetwork } from "@solana/wallet-adapter-base";
import { ConnectionProvider, WalletProvider } from "@solana/wallet-adapter-react";
import { getPhantomWallet } from "@solana/wallet-adapter-wallets";
import { clusterApiUrl } from "@solana/web3.js";
import { FC, ReactNode, useMemo } from "react";

console.warn('phantom', getPhantomWallet)

export const WalletConnectionProvider: FC<{ children: ReactNode }> = ({ children }) => {
    const network = WalletAdapterNetwork.Devnet;
    const endpoint = useMemo(() => clusterApiUrl(network), [network]);

    const wallets = useMemo(
        () => [
            getPhantomWallet()
        ],
    []);

    return (
        <ConnectionProvider endpoint={endpoint}>
            <WalletProvider wallets={wallets} autoConnect>
                {children}
            </WalletProvider>
        </ConnectionProvider>
    )
};
```

Running into `TypeError: (0 , _solana_wallet_adapter_wallets__WEBPACK_IMPORTED_MODULE_3__.getPhantomWallet) is not a function`

After running everything through `next-transpile-modules`.

I'm on next.js `12.0.3` and node `16.4.2`

**Comment by jordansexton** [comment]
Thanks @dylf -- can you post or link a gist to your Webpack / Next config / package.json? Seems like a valid problem but I don't know how to reproduce it. It doesn't look like the issue is with the code importing or using the library, but something about the project / build config.

**Comment by dylf** [comment]
Here is a repo that I get the issue in https://github.com/dylf/solwallet-testing

**Comment by jordansexton** [comment]
Thank you! Will investigate today.

**Comment by jordansexton** [comment]
Okay, after investigating at great length, this is pretty bizarre.

First off, this fails, just as it does in the reproduction you provided:

```tsx
import { getPhantomWallet } from "@solana/wallet-adapter-wallets";
// ...
const wallets = useMemo(() => [getPhantomWallet()], []);
```

If we import the whole package:
```tsx
import * as allWallets from "@solana/wallet-adapter-wallets";

console.log('all wallets', allWallets);
```

We see it's there, as getters:
<img width="356" alt="Screen Shot 2021-11-14 at 1 22 13 PM" src="https://user-images.githubusercontent.com/1173161/141695415-d6465664-8fb5-45f0-9898-0b57d581d759.png">

But in the terminal running `yarn dev`, I see this:
```
all wallets Object [Module] {
  getBitKeepWallet: [Getter],
  getBitpieWallet: [Getter]
}
```

We only see these two wallets. Everything is missing including and after Blocto. This is notable, because Blocto is one of the few wallets with an external dependency -- `@blocto/sdk` -- which is seen in our next.config.js.

Attempting to use Blocto fails with the same error:
```tsx
import { getBloctoWallet } from "@solana/wallet-adapter-wallets";
// ...
const wallets = useMemo(() => [getBloctoWallet()], []);
```

But using BitKeep and Bitpie succeeds:
```tsx
import { getBitKeepWallet, getBitpieWallet } from "@solana/wallet-adapter-wallets";
// ...
const wallets = useMemo(() => [getBitKeepWallet(), getBitpieWallet()], []);
```
<img width="336" alt="Screen Shot 2021-11-14 at 1 29 03 PM" src="https://user-images.githubusercontent.com/1173161/141695591-7db718f2-c541-4672-af15-749f0289aeb6.png">

It seems like loading or transpiling the Blocto SDK is silently failing, and it's causing everything after it to fail. 

I think we could be seeing a side effect of https://github.com/portto/blocto-sdk/issues/14 / https://github.com/portto/blocto-sdk/pull/15.

Even after this issue was closed / PR merged, I'm not convinced that Blocto's SDK works properly when `window` is undefined.

The nextjs-starter package here uses dynamic loading to skip SSR: https://github.com/solana-labs/wallet-adapter/blob/a883ebc5706f38235ee4d5838c2ddb458accdebf/packages/starter/nextjs-starter/pages/_app.tsx#L10-L18

The example package does the same thing: https://github.com/solana-labs/wallet-adapter/blob/a883ebc5706f38235ee4d5838c2ddb458accdebf/packages/starter/example/pages/_app.tsx#L12-L17

This was originally done to sidestep a few wallets unsafely using `window`, `localStorage`, and other Window APIs.

So, as for a workaround for Next.js users -- it's not ideal, but for now, try to use dynamic loading like this.

Figuring out what's actually going on is going to require digging around in the Blocto SDK.

**Comment by dylf** [comment]
Appreciate you digging into this, I'll go with the workaround in the meantime.

**Comment by jordansexton** [comment]
The Blocto SDK has had several updates since this, which may have fixed the issue. Additionally, https://github.com/solana-labs/wallet-adapter/pull/195 has led to a substantial refactor -- you no longer need the `wallets` package, and you can import only the wallet packages you want to use.

**Comment by jordansexton** [comment]
I've checked the Blocto SDK code and they seem to guard all their references to `window` now. In this library we also now guard against using `localStorage` when it isn't defined. I'll close this for now, but please comment if there are still issues.

**Comment by jordansexton** [comment]
https://github.com/solana-labs/wallet-adapter/issues/204 was implemented, which should allow wallets to be safely used in an SSR context because their potentially unsafe dependencies aren't loaded until `connect` is called.

**Comment by bjar-paella** [comment]
I am getting errors working with the next-starter app. 

Here is my package.json 
```
{
  "name": "my-app",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@solana/wallet-adapter-base": "^0.8.1",
    "@solana/wallet-adapter-react": "^0.14.0",
    "@solana/wallet-adapter-react-ui": "^0.7.0",
    "@solana/wallet-adapter-wallets": "^0.13.0",
    "@solana/web3.js": "^1.20.0",
    "next": "^11.1.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@types/react": "^17.0.24",
    "eslint": "^7.32.0",
    "eslint-config-next": "^11.1.2",
    "next-transpile-modules": "^8.0.0",
    "shx": "^0.3.3",
    "typescript": "^4.4.4"
  }
}
```
and my next.config.js
```
const withTM = require("next-transpile-modules")([
  "@blocto/sdk",
  "@project-serum/sol-wallet-adapter",
  "@solana/wallet-adapter-base",
  "@solana/wallet-adapter-react",
  "@solana/wallet-adapter-react-ui",
  "@solana/wallet-adapter-bitpie",
  "@solana/wallet-adapter-blocto",
  "@solana/wallet-adapter-clover",
  "@solana/wallet-adapter-coin98",
  "@solana/wallet-adapter-coinhub",
  "@solana/wallet-adapter-ledger",
  "@solana/wallet-adapter-mathwallet",
  "@solana/wallet-adapter-phantom",
  "@solana/wallet-adapter-safepal",
  "@solana/wallet-adapter-slope",
  "@solana/wallet-adapter-solflare",
  "@solana/wallet-adapter-sollet",
  "@solana/wallet-adapter-solong",
  "@solana/wallet-adapter-tokenpocket",
  "@solana/wallet-adapter-torus",
]);

/** @type {import('next').NextConfig} */
module.exports = withTM({
  reactStrictMode: true,
  webpack5: true,
});
```

This compiles fine, but produces the following module not found error
```
error - unhandledRejection: Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/....../node_modules/@solana/wallet-adapter-bitkeep/lib/adapter' imported from /Users/...../node_modules/@solana/wallet-adapter-bitkeep/lib/index.js
```

**Comment by jordansexton** [comment]
@bjar-paella It looks like you need to run `yarn install` and `yarn build` from the root project directory before running `yarn dev` in the Next.js starter directory.

**Comment by jordansexton** [comment]
@bjar-paella Oh, nevermind -- you're missing some dependencies you need to mark. See https://github.com/solana-labs/wallet-adapter/blob/c118a6597db711e8a1f89f3f47accd4de6d5aaf6/packages/starter/example/next.config.js#L13-L40

**Comment by bjar-paella** [comment]
I see, thanks for the tip. I was just trying to use the package without building it from source. Is that necessary? 

Steps I took: 
1. Created my own Next.JS project 
2. Copied files form the `next-starter` dir
 - next.config.js
 - .tsconfig
 - pages/_app.tsx

I still run into the error with `@solana/wallet-adapter-bitkeep` in the next.config.js. 

Additionally, when I copy all of the dependencies as you show above, I don't have a lot of them installed, i.e. `@solana/wallet-adapter-angular` should I remove these from next.js.config. My apologies for the ignorance.

**Comment by jordansexton** [comment]
I would probably recommend copying the whole next-js-starter directory. It uses more than just _app.tsx.

As for the references to dependencies you don't use in the config, it's okay to leave them in. They don't cause the code to be transpiled when you don't use import them.

**Comment by bjar-paella** [comment]
@jordansexton thanks for the help! I remedied the issues follow these steps:

```sh
git clone https://github.com/solana-labs/wallet-adapter/
cd wallet-adapter
yarn && yarn build
cd packages/starter/nextjs-starter
yarn && yarn build
yarn dev
``` 

then I just copied the started directory out to my own project 
```sh
cp wallet-adapter/packages/starter/next-starter /path/to/project
yarn && yarn build
```
To get this to build without errors I had had to remove dependencies that I had not installed from the next.config.js. I ended up with the following dependencies in my config
```js
const withTM = require("next-transpile-modules")([
    "@blocto/sdk",
    "@project-serum/sol-wallet-adapter",
    "@solana/wallet-adapter-base",
    "@solana/wallet-adapter-react",
    "@solana/wallet-adapter-react-ui",
    "@solana/wallet-adapter-bitkeep",
    "@solana/wallet-adapter-bitpie",
    "@solana/wallet-adapter-blocto",
    "@solana/wallet-adapter-clover",
    "@solana/wallet-adapter-coin98",
    "@solana/wallet-adapter-coinhub",
    "@solana/wallet-adapter-ledger",
    "@solana/wallet-adapter-mathwallet",
    "@solana/wallet-adapter-phantom",
    "@solana/wallet-adapter-safepal",
    "@solana/wallet-adapter-slope",
    "@solana/wallet-adapter-solflare",
    "@solana/wallet-adapter-sollet",
    "@solana/wallet-adapter-solong",
    "@solana/wallet-adapter-tokenpocket",
    "@solana/wallet-adapter-torus",
]);
```

Leaving this here in-case people run into the same errors. Thanks for the help!

**Comment by AppsusUK** [comment]
Hey, I'm using next.config using withTM and everything runs fine using SSR (Thank you!), but I get this error locally and also on Vercel when building but everything still works as intended. Any work arounds to avoid the below?:

```
ReferenceError: localStorage is not defined
--
19:24:48.382 | at /vercel/path0/.next/server/chunks/987.js:1930:21
19:24:48.382 | at Ha (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:26:317)
19:24:48.382 | at Object.useState (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:28:301)
19:24:48.382 | at exports.useState (/vercel/path0/node_modules/react/cjs/react.production.min.js:23:281)
19:24:48.382 | at useLocalStorage (/vercel/path0/.next/server/chunks/987.js:1928:35)
19:24:48.382 | at WalletProvider (/vercel/path0/.next/server/chunks/987.js:2008:27)
19:24:48.382 | at d (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
19:24:48.382 | at bb (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
19:24:48.382 | at a.b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
19:24:48.382 | at a.b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
```

**Comment by zhangxiaoshang** [comment]
> Hey, I'm using next.config using withTM and everything runs fine using SSR (Thank you!), but I get this error locally and also on Vercel when building but everything still works as intended. Any work arounds to avoid the below?:
> 
> ```
> ReferenceError: localStorage is not defined
> --
> 19:24:48.382 | at /vercel/path0/.next/server/chunks/987.js:1930:21
> 19:24:48.382 | at Ha (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:26:317)
> 19:24:48.382 | at Object.useState (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:28:301)
> 19:24:48.382 | at exports.useState (/vercel/path0/node_modules/react/cjs/react.production.min.js:23:281)
> 19:24:48.382 | at useLocalStorage (/vercel/path0/.next/server/chunks/987.js:1928:35)
> 19:24:48.382 | at WalletProvider (/vercel/path0/.next/server/chunks/987.js:2008:27)
> 19:24:48.382 | at d (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
> 19:24:48.382 | at bb (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
> 19:24:48.382 | at a.b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
> 19:24:48.382 | at a.b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
> ```

I have the same problem

**Comment by jordansexton** [comment]
Thanks for the report. It's a warning caused by some code that accesses `localStorage` on the server where it doesn't exist. I'll add a guard to silence this, it shouldn't effect how it works.

**Comment by systemis** [comment]
Hi @jordansexton, I got this error when interage wallet adapter in react typescript app: 
`Compiled with problems:X

ERROR in ./src/app.tsx 20:37-57

export 'PhantomWalletAdapter' (imported as 'PhantomWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 20:65-83

export 'SlopeWalletAdapter' (imported as 'SlopeWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 20:91-112

export 'SolflareWalletAdapter' (imported as 'SolflareWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 20:120-138

export 'TorusWalletAdapter' (imported as 'TorusWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 20:146-165

export 'LedgerWalletAdapter' (imported as 'LedgerWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 20:173-192

export 'SolletWalletAdapter' (imported as 'SolletWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 22:10-38

export 'SolletExtensionWalletAdapter' (imported as 'SolletExtensionWalletAdapter') was not found in '@solana/wallet-adapter-wallets' (module has no exports)


ERROR in ./src/app.tsx 25:30-48

export 'ConnectionProvider' (imported as 'ConnectionProvider') was not found in '@solana/wallet-adapter-react' (module has no exports)


ERROR in ./src/app.tsx 27:35-49

export 'WalletProvider' (imported as 'WalletProvider') was not found in '@solana/wallet-adapter-react' (module has no exports)


ERROR in ./node_modules/@solana/wallet-adapter-bitkeep/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-bitkeep/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-bitpie/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-bitpie/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-blocto/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-blocto/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-clover/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-clover/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-coinhub/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-coinhub/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-coin98/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-coin98/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.


ERROR in ./node_modules/@solana/wallet-adapter-ledger/lib/esm/index.js 1:0-26

Module not found: Error: Can't resolve './adapter' in '/Users/thinh/Desktop/work/kyros/mint-nft/node_modules/@solana/wallet-adapter-ledger/lib/esm'
Did you mean 'adapter.js'?
BREAKING CHANGE: The request './adapter' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
`
<img width="1451" alt="Ảnh chụp Màn hình 2022-01-02 lúc 16 07 08" src="https://user-images.githubusercontent.com/19547873/147871271-8e131fe6-b039-4058-8d61-6195410bffae.png">


Here is my package.json file
`{
  "name": "mintnft",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^4.12.3",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/lab": "^4.0.0-alpha.60",
    "@project-serum/sol-wallet-adapter": "^0.2.6",
    "@solana/wallet-adapter-base": "0.9.0",
    "@solana/wallet-adapter-react": "0.15.0",
    "@solana/wallet-adapter-react-ui": "0.8.0",
    "@solana/wallet-adapter-wallets": "0.14.0",
    "@solana/web3.js": "^1.31.0",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.0.3",
    "@types/node": "^16.11.17",
    "@types/react": "^17.0.38",
    "@types/react-dom": "^17.0.11",
    "bootstrap": "^5.1.3",
    "cross-env": "^7.0.3",
    "crypto": "^1.0.1",
    "crypto-js": "^4.1.1",
    "history": "5",
    "lodash": "^4.17.21",
    "react": "^17.0.2",
    "react-countdown": "^2.3.2",
    "react-dom": "^17.0.2",
    "react-redux": "^7.2.6",
    "react-router-dom": "6",
    "react-scripts": "5.0.0",
    "reactstrap": "^9.0.1",
    "redux": "^4.1.2",
    "redux-saga": "^1.1.3",
    "sass": "^1.45.1",
    "styled-components": "^5.3.1",
    "typescript": "^4.5.4",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@types/lodash": "^4.14.178",
    "dotenv": "^10.0.0",
    "eslint": "^8.5.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "prettier": "^2.5.1"
  },
  "browser": {
    "crypto": false,
    "stream": false
  },
  "lint": "eslint .",
  "lint:fix": "eslint --fix",
  "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
}
`
